src/erlfmt_format.erl

```diff
@@ -311,10 +311,18 @@ binary_op_to_algebra(Op, Meta, Left, Right, Indent) ->
     IsNextBreakFits = is_next_break_fits_op(Op, Right) andalso not HasBreak,
 
     Doc =
-        with_next_break_fits(IsNextBreakFits, RightD, fun (R) ->
-            RightOpD = nest(break(concat(<<" ">>, OpD), group(R)), Indent, break),
-            concat(group(LeftD), group(concat(maybe_force_breaks(HasBreak), RightOpD)))
-        end),
+        case Op of
+            '|' ->
+                 with_next_break_fits(IsNextBreakFits, RightD, fun (R) ->
+                    RightOpD = nest(break(concat(<<" ">>, OpD), R), Indent, break),
+                    concat(LeftD, concat(maybe_force_breaks(HasBreak), RightOpD))
```

Comment:
I'm not sure this would behave correctly with mixed operators. For example something like `1 + 2 | 3 + 4`

Reply:
This seems to be a parse error `"1 + 2 | 3 + 4\n"` and `"Foo = 1 + 2 | 3 + 4\n"`
I didn't know you could write that, do you have a parsable example?

Reply:
I'm sorry. This should be a complete example that parses fine today:
```erl
-type foo() :: 1 + 2 | 3 + 4.
```

Reply:
These both work
```erlang
?assertSame("-type foo() :: 1 + 2 | 3 + 4.\n"),
    ?assertSame(
        "-type foo() ::\n"
        "    1 + 2 |\n"
        "    3 + 4.\n"
    ).
```

Reply:
I could add this one as a test
```erlang
?assertSame(
    "-type foo() ::\n"
    "    1 + 2 |\n"
    "    3 + 4.\n"
).
```
But maybe you have a different test in mind?

Reply:
Looks good. It seems the thing I was worried about works correctly. Great job!

Reply:
I got Lucky :)